src/main/scala/com/lucidchart/android/lifecycle/LifecycleManaged.scala

```diff
@@ -37,8 +37,22 @@ class LifecycleMacro(val c: Context) {
           }
         """ :: maybeCompanion
 
+      case q"""
+        $mods trait $tpname[..$tparams] extends { ..$earlydefns  } with ..$parents { $self =>
+          ..$members
+        }
+      """ :: maybeCompanion =>
+        val ActionValidationResult(membersWithExpandedActions, actions) = validateActions(members, parameters)
```

Comment:
Looks like you could just make a method for both the trait and class that does both these actions.